[IMP] l10n_es: cleanup some template taxes

yvsg-odoo opened the pull request:
Changes for taxes in template data were requested because they were confusing or new taxes needed to be created.
List of changes applied: https://www.odoo.com/web#id=3395186&menu_id=4720&cids=1&action=333&active_id=967&model=project.task&view_type=form

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/127521).

chklop commented:
@rafaelbn FYI !

rafaelbn commented:
Hello @chklop , I would like to test this PR in runbot but it looks like not available, could you please rebuild runbot to test this PR? MAybe you could rebase @yvsg-odoo ?

Thank you!

rafaelbn commented:
@jco-odoo could you help in fixing runbot of this PR to test it?

rafaelbn commented:
@C3POdoo fix runbot please ❤️ 

@mart-e @jco-odoo  please do you know how to rebuild this PR in order to test it in runbot? When I try it looks like the runbot is "off"

I just want to test it! This is a really important for a better world change! ❤️

mart-e commented:
Hey Rafael :wave: 

There is nothing wrong with the runbot, the build was killed after some time (we have hundreds of build each day).
The problem is that there is a conflict in the files now `addons/l10n_es_edi_sii/data/template/account.tax-es_common.csv` contains translatins since 5d9ee9f0d2c1079f876945493a93669241d17b37

@yvsg-odoo could you check if this is still needed and resolve the conflict if this is the case?

yvsg-odoo commented:
Ok, I'll be resolving this asap, thanks!

h4818 commented on addons/l10n_es/data/template/account.fiscal.position-es_common.csv:
```diff
@@ -61,7 +61,7 @@
 "","","","","","","","","","","account_common_7040","account_common_7042"
 "","","","","","","","","","","account_common_7050","account_common_7052"
-"fp_not_subject_tai","","Regime not subject to localition rules (TAI, Canarias, ...)","Régimen No sujeto por reglas de localización (TAI - Canarias, Ceuta, Melilla...)","","","","","account_tax_template_p_iva4_bc","account_tax_template_p_iva0_ns_b","",""
+"fp_not_subject_tai","","Régimen No sujeto por reglas de localización (TAI)","Régimen No sujeto por reglas de localización (TAI - Canarias, Ceuta, Melilla...)","","","","","account_tax_template_p_iva4_bc","account_tax_template_p_iva0_ns_b","",""
```
would it not make sense to keep (but fix) the English, since we have the translated term

rafaelbn reviewed (changes requested):
1. Go to runbot -> Apps
2. Try to install `l10n_es_edi_sii` Spain - SII EDI Suministro de Libros
3. You the this error

![imagen](https://github.com/odoo/odoo/assets/8736623/d7bbe527-06ac-42ce-8b3a-69db71c84951)

Could you please take a look?

Thank you!

rafaelbn reviewed (changes requested):
1. Create a new company with Spain as country
2. Go to invoicing -> Configuration and try to set the Fiscal Packaget
3. You will get:

```
RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/data/build/odoo/odoo/http.py", line 1710, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/data/build/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/data/build/odoo/odoo/http.py", line 1737, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/data/build/odoo/odoo/http.py", line 1938, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/data/build/odoo/addons/website/models/ir_http.py", line 233, in _dispatch
    response = super()._dispatch(endpoint)
  File "/data/build/odoo/odoo/addons/base/models/ir_http.py", line 191, in _dispatch
    result = endpoint(**request.params)
  File "/data/build/odoo/odoo/http.py", line 717, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/data/build/odoo/addons/web/controllers/dataset.py", line 34, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/data/build/odoo/addons/web/controllers/dataset.py", line 26, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/data/build/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/data/build/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/data/build/odoo/odoo/addons/base/models/res_config.py", line 602, in execute
    self.set_values()
  File "/data/build/odoo/addons/website_sale_stock/models/res_config_settings.py", line 26, in set_values
    super(ResConfigSettings, self).set_values()
  File "/data/build/odoo/addons/sale_project/models/res_config_settings.py", line 11, in set_values
    super().set_values()
  File "/data/build/odoo/addons/website_sale/models/res_config_settings.py", line 68, in set_values
    super().set_values()
  File "/data/build/odoo/addons/website_payment_authorize/models/res_config_settings.py", line 25, in set_values
    super().set_values()
  File "/data/build/odoo/addons/sale_management/models/res_config_settings.py", line 31, in set_values
    super().set_values()
  File "/data/build/odoo/addons/sale/wizard/res_config_settings.py", line 94, in set_values
    super().set_values()
  File "/data/build/odoo/addons/point_of_sale/models/res_config_settings.py", line 157, in set_values
    super(ResConfigSettings, self).set_values()
  File "/data/build/odoo/addons/crm_iap_enrich/models/res_config_settings.py", line 18, in set_values
    super().set_values()
  File "/data/build/odoo/addons/stock_account/models/res_config_settings.py", line 18, in set_values
    super().set_values()
  File "/data/build/odoo/addons/purchase/models/res_config_settings.py", line 41, in set_values
    super().set_values()
  File "/data/build/odoo/addons/mrp/models/res_config_settings.py", line 27, in set_values
    super().set_values()
  File "/data/build/odoo/addons/hr_expense/models/res_config_settings.py", line 36, in set_values
    super().set_values()
  File "/data/build/odoo/addons/stock/models/res_config_settings.py", line 96, in set_values
    super().set_values()
  File "/data/build/odoo/addons/project/models/res_config_settings.py", line 70, in set_values
    super().set_values()
  File "/data/build/odoo/addons/mass_mailing/models/res_config_settings.py", line 36, in set_values
    super().set_values()
  File "/data/build/odoo/addons/crm/models/res_config_settings.py", line 135, in set_values
    super(ResConfigSettings, self).set_values()
  File "/data/build/odoo/addons/account/models/res_config_settings.py", line 199, in set_values
    self.env['account.chart.template'].try_loading(self.chart_template, company=self.company_id)
  File "/data/build/odoo/addons/account/models/chart_template.py", line 142, in try_loading
    return self._load(template_code, company, install_demo)
  File "/data/build/odoo/addons/point_of_sale/models/chart_template.py", line 22, in _load
    result = super()._load(template_code, company, install_demo)
  File "/data/build/odoo/addons/account/models/chart_template.py", line 193, in _load
    self._post_load_data(template_code, company, template_data)
  File "/data/build/odoo/addons/stock_account/models/account_chart_template.py", line 12, in _post_load_data
    super()._post_load_data(template_code, company, template_data)
  File "/data/build/odoo/addons/account/models/chart_template.py", line 509, in _post_load_data
    self._setup_utility_bank_accounts(template_code, company, template_data)
  File "/data/build/odoo/addons/account/models/chart_template.py", line 652, in _setup_utility_bank_accounts
    accounts = self.env['account.account'].create(accounts_data.values())
  File "<decorator-gen-255>", line 2, in create
  File "/data/build/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/data/build/odoo/addons/account/models/account_account.py", line 727, in create
    vals['code'] = self._search_new_account_code(company, vals.pop('code_digits'), vals.pop('prefix'), cache)
  File "/data/build/odoo/addons/account/models/account_account.py", line 356, in _search_new_account_code
    new_code = str(prefix.ljust(digits - 1, '0')) + str(num)
AttributeError: 'bool' object has no attribute 'ljust'

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    RPCError@https://49054354-master-all.runbot179.odoo.com/web/assets/1960-84863e7/web.assets_backend.min.js:1344:319
    makeErrorFromResponse@https://49054354-master-all.runbot179.odoo.com/web/assets/1960-84863e7/web.assets_backend.min.js:1347:163
    jsonrpc/promise</<@https://49054354-master-all.runbot179.odoo.com/web/assets/1960-84863e7/web.assets_backend.min.js:1351:34
```

rafaelbn commented:
Please ping me when I can test again 😄 🙏🏼 . Thank you!

h4818 commented on addons/l10n_es_edi_sii/data/template/account.tax-es_common.csv:
```diff
@@ -1,94 +1,97 @@
+"account_tax_template_p_iva4_bc","4% IVA soportado (bienes corrientes)","","","sujeto",False,"","consu",""
+"account_tax_template_p_iva10_sc","10% IVA soportado (servicios corrientes)","","","sujeto",False,"","consu",""
+"account_tax_template_s_iva0","IVA Exento (Otros)","","","exento",False,"E6","",""
+"account_tax_template_s_iva0_art_22","IVA Exento (Art. 22)","","","exento",False,"E3","","False"
```
`active` is boolean

yvsg-odoo commented:
Hi @rafaelbn could you please check again if the errors persist in the new build and if so, help me reproduce them? I wasn't able to reproduce them but I wasn't aiming to fix them yet! Thanks 😄

rafaelbn commented:
@yvsg-odoo , catch it! I can reproduce but is a really staing use case. Only happend if you try to install sii before setup the localization package in the company

https://www.loom.com/share/6fb1ec29fff54a8b8cf2650869f50a08?sid=7c332c3a-de86-4eae-b0be-3eb9bb90474e

In the other hand, why all the names for taxes are wrong? All have change the name vs v16

Are you aware of it? We should fix it

![imagen](https://github.com/odoo/odoo/assets/8736623/e83fe99d-d6a8-4f16-b8f9-cfaef72195e4)

yvsg-odoo commented:
@rafaelbn Sorry for the delay. Thanks for the video. I could reproduce the error by creating a spanish company before trying to install. But I could also reproduce it on master. It's good to double check, but if I tested correctly it isn't a problem from this PR

jco-odoo commented:
@rafaelbn The idea is still to have short names for the taxes themselves.  Hence the extra fields, ...  However, it needs to be improved.

rafaelbn commented:
Can I send you an Excel as a proposition? Actually they are in English! obviously they are not valid.

jco-odoo commented:
@rafaelbn Yes, that would be really helpful that you edit the CSV according to what you think is best.  There is however an idea behind the short names: it allows for speed encoding things so you can quickly type something which gives you the right tax, but the changes might have been too hasty and I think there is a need for a name@es column as well.  I think short names like BI for bienes de inversion and BC for bienes corrientes might be clearer.  The idea is to make it as generic as possible, but it should still be clear...

cc @chklop

yvsg-odoo commented:
@rafaelbn @jco-odoo Hello, about the Excel, should we apply these changes in this PR? What do you guys think? Because these aren't changes from this PR as well as the errors pointed in the video. Maybe it's valid to create a separate task (and I could also take it if it's the case, just would validate with @rugo-odoo )

jco-odoo commented:
@yvsg-odoo Ok to keep the changes here to a minimum and to do the others in another PR.  I would still just remove those columns in l10n_es_edi_sii.

yvsg-odoo commented:
Hi @jco-odoo  , thanks for the feedback. I've removed description and invoice_label columns. Kept name to prevent errors. I've also fixed the "No sujeto" label. Do you think it's ok for us to proceed now?

jco-odoo commented on addons/l10n_es/data/template/account.fiscal.position-es_common.csv:
```diff
@@ -61,7 +61,7 @@
 "","","","","","","","","","","account_common_7040","account_common_7042"
 "","","","","","","","","","","account_common_7050","account_common_7052"
-"fp_not_subject_tai","","Regime not subject to localition rules (TAI, Canarias, ...)","Régimen No sujeto por reglas de localización (TAI - Canarias, Ceuta, Melilla...)","","","","","account_tax_template_p_iva4_bc","account_tax_template_p_iva0_ns_b","",""
+"fp_not_subject_tai","","Regime not subject to localization rules (TAI - Canarias, Ceuta, Melilla...)","Régimen No sujeto por reglas de localización (TAI - Canarias, Ceuta, Melilla...)","","","","","account_tax_template_p_iva4_bc","account_tax_template_p_iva0_ns_b","",""
```
```suggestion
"fp_not_subject_tai","","Regime not subject to localization rules","Régimen No sujeto por reglas de localización","","","","","account_tax_template_p_iva4_bc","account_tax_template_p_iva0_ns_b","",""
```

jco-odoo commented on addons/l10n_es_edi_sii/data/template/account.tax-es_common.csv:
```diff
@@ -1,94 +1,97 @@
-"account_tax_template_p_rp19","Retenciones 19% (préstamos)","","","retencion","","",""
-"account_tax_template_p_rrD19","Retenciones 19% (reparto de dividendos)","","","retencion","","",""
+"id","name","l10n_es_type","l10n_es_bien_inversion","l10n_es_exempt_reason","tax_scope","active"
+"account_tax_template_s_iva21b","","sujeto",False,"","consu",True
```
Here the name is empty, while in later records it is the Spanish instead of the English.  So, better to remove it entirely.

yvsg-odoo replied:
When I remove the name column I get a database error saying name is mandatory. But I could leave them all empty

jco-odoo commented on addons/l10n_es/data/template/account.tax-es_common.csv:
```diff
@@ -213,11 +213,11 @@
 "","","","","","","","","","","","","base","","refund","","",""
 "","","","","","","","","","","","","tax","","refund","account_common_472","",""
-"account_tax_template_s_iva0_ns","","VAT Exempt not subjected","VAT Exempt Non-Subject","sale","0% EXEMPT NS","IVA Exento Repercutido No Sujeto","0.0","percent","tax_group_iva_0","","","base","","invoice","","","IVA Exento No Sujeto"
+"account_tax_template_s_iva0_ns","","VAT Exempt not subjected","VAT Exempt Non-Subject","sale","0% EXEMPT NS","IVA Exento Sin Derecho A Deducción","0.0","percent","tax_group_iva_0","","","base","","invoice","","","IVA Exento Sin Derecho A Deducción"
```
This tax is ignored when sending, so it should be "IVA no aplicable" in Spanish and VAT Not Applicable in English.

chklop replied:
@rafaelbn I tend to agree with @jco-odoo remark here.

yvsg-odoo replied:
I didn't change "0% EXEMPT NS" yet, is it also relevant to change?

rafaelbn replied:
We agree [here](https://github.com/odoo/odoo/issues/124942#issuecomment-1618655448)

> Tax l10n_es.account_tax_template_s_iva0_ns | "IVA Exento Repercutido No Sujeto"
> ​=> Rename to IVA Exento sin derecho a deducción

Is it @chklop @jco-odoo ?

rafaelbn replied:
> Tax l10n_es.account_tax_template_s_iva0_ns | "IVA Exento Repercutido No Sujeto"
​=> Rename to IVA Exento sin derecho a deducción

@jco-odoo because is Art. 20 and is "Casilla 83" in 303. I showed you this:

![imagen](https://github.com/odoo/odoo/assets/8736623/bbb7c1cf-bbfc-4a6c-a092-40addce3d6a5)

Ref: https://sede.agenciatributaria.gob.es/Sede/todas-gestiones/impuestos-tasas/iva/modelo-303-iva-autoliquidacion_/instrucciones.html

jco-odoo replied:
@rafaelbn Thanks again!  We should add the tag for the casilla 83 as well then.

rafaelbn replied:
> Thanks again! We should add the tag for the casilla 83 as well then.

@jco-odoo no, remember (Valencia is far away 😄 , because casilla 83 is only for the LAST period of the year and not quarterly or monthly. This is a really special use case.

jco-odoo replied:
@rafaelbn I remember that.  It is just that I thought the report would need to filter it out.  (based on the period)

jco-odoo commented on addons/l10n_es/data/template/account.tax-es_common.csv:
```diff
@@ -369,7 +369,7 @@
 "","","","","","","","","","","","","base","","refund","","",""
 "","","","","","","","","","","","","tax","l10n_es.mod_303_42","refund","account_common_472","",""
-"account_tax_template_s_iva0_e","","VAT 0% export (goods)","Export (Goods)","sale","0% EX G","IVA 0% Exportaciones","0.0","percent","tax_group_iva_0","","","base","l10n_es.mod_303_60","invoice","","","Exportación (Bienes)"
+"account_tax_template_s_iva0_e","","VAT 0% export (goods)","Export (Goods)","sale","0% EX G","IVA Exento Exportaciones","0.0","percent","tax_group_iva_0","","","base","l10n_es.mod_303_60","invoice","","","Exportación (Bienes)"
```
Spanish and English are not consistent.

jco-odoo reviewed (commented):
Would be good to check if a rebase works or not as well.

yvsg-odoo commented:
@jco-odoo I tried to adapt checking your PR https://github.com/odoo/odoo/pull/124971/files and I believe it's now ok. Can you please check and then we can see what's left to do so we can proceed

jco-odoo commented on addons/l10n_es/data/template/account.tax-es_common.csv:
```diff
@@ -213,11 +213,15 @@
+"","","","","","","","","","","","","tax","","invoice","","",""
+"","","","","","","","","","","","","base","","refund","","",""
+"","","","","","","","","","","","","tax","","refund","","",""
+"account_tax_template_s_iva0_no_aplicable","","VAT Not Applicable","VAT Not Applicable","sale","0% NOT APPLICABLE","IVA No Aplicable","0.0","percent","tax_group_iva_0","","","base","","invoice","","","IVA No Aplicable"
```
this one should appear in l10n_es_edi_sii as ignore

qdp-odoo commented:
@jco-odoo are we waiting anything from you on this task, or can we merge it like that?

jco-odoo commented:
@qdp-odoo It seems fine.  I wanted to go through one last time.

jco-odoo commented:
@robodoo r+

robodoo commented:
@yvsg-odoo @jco-odoo staging failed: ci/runbot (view more at https://runbot.odoo.com/runbot/build/50815409)

FlorianGilbert commented:
@robodoo retry

jco-odoo commented:
It should fail staging because of a test.  Things should be done a bit differently anyways.

jco-odoo commented:
@malb-odoo I can still continue in this branch.

malb-odoo commented:
@jco-odoo I will put you has the dev on the task then, thanks !

jco-odoo commented:
Doing it in https://github.com/odoo/odoo/pull/139514 anyways.

malb-odoo commented:
Then let's close this one right ?
